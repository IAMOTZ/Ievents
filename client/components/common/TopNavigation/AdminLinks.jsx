import React from 'react';
import { Link } from 'react-router-dom';
import FakeDiv from '../../hoc/Aux';

const AdminLinks = (props) => {
  let component;
  if (props.isAdmin || props.isSuperAdmin) {
    component = (
      <FakeDiv>
        <li className="nav-item">
          <Link to="/addCenter" className="nav-link">Add Center</Link>
        </li>
        {
          props.isSuperAdmin ?
            <li className="nav-item">
              <Link to="/addAdmin" className="nav-link">Add Admin</Link>
            </li> : null
        }
        <li className="nav-item">
          <Link to="/transactions" className="nav-link">Transactions</Link>
        </li>
      </FakeDiv>
    );
  } else {
    component = null;
  }
  return component;
};

export default AdminLinks;
