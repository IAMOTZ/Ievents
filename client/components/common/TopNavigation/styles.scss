$custom-black: rgba(0,0,0,0.9);
$custom-blue: rgb(0, 142, 214);

@mixin nav-bar-theme {
  background-color: $custom-black;
  border-bottom: 3px solid $custom-blue;
}

nav#regular-top-navbar{
  @include nav-bar-theme;
  .nav-link:hover {
    color: white;
  }
  .nav-item {
    margin-left: 10px;
    margin-right: 10px;
    &:last-child {
      margin-right: 0px;
    }
    &:first-child {
      margin-left: 0px;
    }
  }
}

nav#auth-top-navbar {
  @include nav-bar-theme;
}