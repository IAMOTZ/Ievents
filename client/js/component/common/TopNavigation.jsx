import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';

import { clearUser } from '../../actions/authAction';

@connect((store) => {
  return {

  }
})

// The navigation that is specific to user page
class UserTopNav extends React.Component {
  logout = () => {
    this.props.dispatch(clearUser());
  }
  render() {
    return (
      <div>
        <nav class="navbar navbar-expand-sm navbar-dark fixed-top d-block d-lg-none">
          <div class="container">
            <a href="" class="navbar-brand">
              <strong class="text-capitalize">{this.props.name}</strong>
              <br />
              <p class="h6">{this.props.title}</p>
            </a>
            <button class="navbar-toggler" data-toggle="collapse" data-target="#navMenu">
              <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="navMenu">
              <ul class="navbar-nav ml-auto">
                <li class="nav-item">
                  <Link to="/events" className="nav-link">My events</Link>
                </li>
                <li class="nav-item">
                  <Link to="/addEvent" className="nav-link">Add Events</Link>
                </li>
                <li class="nav-item">
                  <Link to="#" className="nav-link">Centers</Link>
                </li>
                <li class="nav-item">
                  <Link to="/" className="nav-link" onClick={this.logout}>Logout</Link>
                </li>
              </ul>
            </div>
          </div>
        </nav>
      </div>
    );
  }
}

// The navigation that is returned by default
export default () => {
  return (
    <nav className="navbar navbar-expand-sm navbar-dark fixed-top">
      <div className="container">
        <Link to="/" className="navbar-brand">
          <strong>Ievents</strong>
        </Link>
        <button className="navbar-toggler" data-toggle="collapse" data-target="#navMenu">
          <span className="navbar-toggler-icon" />
        </button>
        <div className="collapse navbar-collapse" id="navMenu">
          <ul className="navbar-nav ml-auto">
            <li className="nav-item">
              <Link to="/" className="nav-link">Home</Link>
            </li>
            <li className="nav-item">
              <Link to="#" className="nav-link">Centers</Link>
            </li>
            <li className="nav-item">
              <Link to="/users/login" className="nav-link"> Signin </Link>
            </li>
            <li className="nav-item">
              <Link to="/users" className="nav-link"> Signup </Link>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  );
}

// The navigation that is specific to admin page  
const AdminTopNav = (props) => {
  return (
    <div>
      <nav class="navbar navbar-expand-sm navbar-dark fixed-top d-block d-lg-none">
        <div class="container">
          <a href="" class="navbar-brand">
            <strong>Admin</strong>
            <br />
            <p class="h6">{props.title}</p>
          </a>
          <button class="navbar-toggler" data-toggle="collapse" data-target="#navMenu">
            <span class="navbar-toggler-icon"></span>
          </button>
          <div class="collapse navbar-collapse" id="navMenu">
            <ul class="navbar-nav ml-auto">
              <li class="nav-item">
                <Link to="#" class="nav-link">Centers</Link>
              </li>
              <li class="nav-item">
                <Link to="#" class="nav-link">Add Center</Link>
              </li>
              <li class="nav-item">
                <Link to="#" class="nav-link">Transactions</Link>
              </li>
              <li class="nav-item">
                <Link to="#" class="nav-link">Logout</Link>
              </li>
            </ul>
          </div>
        </div>
      </nav>
    </div>
  );
};

export { UserTopNav, AdminTopNav };
