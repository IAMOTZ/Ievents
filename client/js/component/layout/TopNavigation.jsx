import React from 'react';

export default () => {
  return (
    <nav className="navbar navbar-expand-sm navbar-dark fixed-top">
      <div className="container">
        <a href="index.html" className="navbar-brand">
          <strong>Ievents</strong>
        </a>
        <button className="navbar-toggler" data-toggle="collapse" data-target="#navMenu">
          <span className="navbar-toggler-icon" />
        </button>
        <div className="collapse navbar-collapse" id="navMenu">
          <ul className="navbar-nav ml-auto">
            <li className="nav-item">
              <a href="index.html" className="nav-link">Home</a>
            </li>
            <li className="nav-item">
              <a href="centers.html" className="nav-link">Centers</a>
            </li>
            <li className="nav-item">
              <a href="signin.html" className="nav-link">Signin</a>
            </li>
            <li className="nav-item">
              <a href="#" className="nav-link">Signup</a>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  );
};
