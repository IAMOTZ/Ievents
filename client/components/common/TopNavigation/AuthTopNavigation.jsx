import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import './styles.scss';
import AdminLinks from './AdminLinks';
import { logOut } from '../../../actions/authAction';


const AuthTopNavigation = props => (
  <nav className="navbar navbar-expand-md navbar-dark fixed-top d-block d-lg-none" id="auth-top-navbar">
    <div className="container">
      <a href="#" className="navbar-brand">
        <strong className="text-capitalize">{props.name}</strong>
        <br />
        <p className="h6">{props.title}</p>
      </a>
      <button className="navbar-toggler" data-toggle="collapse" data-target="#navMenu">
        <span className="navbar-toggler-icon" />
      </button>
      <div className="collapse navbar-collapse" id="navMenu">
        <ul className="navbar-nav ml-auto">
          <li className="nav-item">
            <Link to="/events" className="nav-link">My events</Link>
          </li>
          <li className="nav-item">
            <Link to="/addEvent" className="nav-link">Create Event</Link>
          </li>
          <li className="nav-item">
            <Link to="/centers2" className="nav-link">Centers</Link>
          </li>
          <AdminLinks isAdmin={props.isAdmin} isSuperAdmin={props.isSuperAdmin} />
          <li className="nav-item">
            <Link to="/profile" className="nav-link">Profile</Link>
          </li>
          <li className="nav-item">
            <Link to="/" className="nav-link" onClick={() => props.dispatch(logOut())}>Logout</Link>
          </li>
        </ul>
      </div>
    </div>
  </nav>
);

AuthTopNavigation.propTypes = {
  name: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  isAdmin: PropTypes.bool.isRequired,
  isSuperAdmin: PropTypes.bool.isRequired,
  dispatch: PropTypes.func.isRequired,
};

export default AuthTopNavigation;
