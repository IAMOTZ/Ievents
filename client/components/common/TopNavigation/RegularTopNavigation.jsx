import React from 'react';
import { Link } from 'react-router-dom';
import './styles.scss';

const RegularTopNavigation = () => (
  <nav className="navbar navbar-expand-sm navbar-dark fixed-top" id="regular-top-navbar">
    <div className="container">
      <Link to="/" className="navbar-brand">
        <strong>iEvents</strong>
      </Link>
      <button className="navbar-toggler" data-toggle="collapse" data-target="#navMenu">
        <span className="navbar-toggler-icon" />
      </button>
      <div className="collapse navbar-collapse" id="navMenu">
        <ul className="navbar-nav ml-auto">
          <li className="nav-item">
            <Link to="/" className="nav-link">Home</Link>
          </li>
          <li className="nav-item">
            <Link to="/explore/centers" className="nav-link">Centers</Link>
          </li>
          <li className="nav-item">
            <Link to="/users/login" className="nav-link"> Log in </Link>
          </li>
          <li className="nav-item">
            <Link to="/users/signup" className="nav-link"> Sign up </Link>
          </li>
        </ul>
      </div>
    </div>
  </nav>
);

export default RegularTopNavigation;
