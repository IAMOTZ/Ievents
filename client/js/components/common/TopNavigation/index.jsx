import RegularTopNavigation from './RegularTopNavigation';
import AuthTopNavigation from './AuthTopNavigation';

export { RegularTopNavigation, AuthTopNavigation };
