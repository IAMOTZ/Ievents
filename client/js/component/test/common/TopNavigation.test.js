/* global shallow */
/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import TopNavigation, { UserTopNav } from '../../common/TopNavigation.jsx';

const props = {
  name: 'test-name',
  isAdmin: false,
  isSuperAdmin: false,
  dispatch: () => { },
};
const alterProps = newProps => ({
  name: props.name,
  isAdmin: Boolean(newProps.isAdmin),
  isSuperAdmin: Boolean(newProps.isSuperAdmin),
  dispatch: props.dispatch,
});

describe('<TopNavigation />', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<TopNavigation />);
    expect(wrapper).toMatchSnapshot();
  });
});
describe('<UserTopNav />', () => {
  it('should render correctly for normal user', () => {
    const wrapper = shallow(<UserTopNav {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render correctly for admin user', () => {
    const wrapper = shallow(<UserTopNav {...alterProps({ isAdmin: true }) } />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render correctly for super admin user', () => {
    const wrapper = shallow(<UserTopNav {...alterProps({ isSuperAdmin: true }) } />);
    expect(wrapper).toMatchSnapshot();
  });
});
