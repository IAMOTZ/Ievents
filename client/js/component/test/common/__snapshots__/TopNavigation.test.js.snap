// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopNavigation /> should render correctly 1`] = `
<nav
  className="navbar navbar-expand-sm navbar-dark fixed-top"
>
  <div
    className="container"
  >
    <Link
      className="navbar-brand"
      replace={false}
      to="/"
    >
      <strong>
        Ievents
      </strong>
    </Link>
    <button
      className="navbar-toggler"
      data-target="#navMenu"
      data-toggle="collapse"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navMenu"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <li
          className="nav-item"
        >
          <Link
            className="nav-link"
            replace={false}
            to="/"
          >
            Home
          </Link>
        </li>
        <li
          className="nav-item"
        >
          <Link
            className="nav-link"
            replace={false}
            to="/centers1"
          >
            Centers
          </Link>
        </li>
        <li
          className="nav-item"
        >
          <Link
            className="nav-link"
            replace={false}
            to="/users/login"
          >
             Signin 
          </Link>
        </li>
        <li
          className="nav-item"
        >
          <Link
            className="nav-link"
            replace={false}
            to="/users"
          >
             Signup 
          </Link>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;

exports[`<UserTopNav /> should render correctly for admin user 1`] = `
<nav
  className="navbar navbar-expand-md navbar-dark fixed-top d-block d-lg-none"
>
  <div
    className="container"
  >
    <a
      className="navbar-brand"
      href=""
    >
      <strong
        className="text-capitalize"
      >
        test-name
      </strong>
      <br />
      <p
        className="h6"
      />
    </a>
    <button
      className="navbar-toggler"
      data-target="#navMenu"
      data-toggle="collapse"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navMenu"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <li
          className="nav-item"
        >
          <Link
            className="nav-link"
            replace={false}
            to="/events"
          >
            My events
          </Link>
        </li>
        <li
          className="nav-item"
        >
          <Link
            className="nav-link"
            replace={false}
            to="/addEvent"
          >
            Add Events
          </Link>
        </li>
        <li
          className="nav-item"
        >
          <Link
            className="nav-link"
            replace={false}
            to="/centers2"
          >
            Centers
          </Link>
        </li>
        <AdminOptions
          isAdmin={true}
          isSuperAdmin={false}
        />
        <li
          className="nav-item"
        >
          <Link
            className="nav-link"
            onClick={[Function]}
            replace={false}
            to="/"
          >
            Logout
          </Link>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;

exports[`<UserTopNav /> should render correctly for normal user 1`] = `
<nav
  className="navbar navbar-expand-md navbar-dark fixed-top d-block d-lg-none"
>
  <div
    className="container"
  >
    <a
      className="navbar-brand"
      href=""
    >
      <strong
        className="text-capitalize"
      >
        test-name
      </strong>
      <br />
      <p
        className="h6"
      />
    </a>
    <button
      className="navbar-toggler"
      data-target="#navMenu"
      data-toggle="collapse"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navMenu"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <li
          className="nav-item"
        >
          <Link
            className="nav-link"
            replace={false}
            to="/events"
          >
            My events
          </Link>
        </li>
        <li
          className="nav-item"
        >
          <Link
            className="nav-link"
            replace={false}
            to="/addEvent"
          >
            Add Events
          </Link>
        </li>
        <li
          className="nav-item"
        >
          <Link
            className="nav-link"
            replace={false}
            to="/centers2"
          >
            Centers
          </Link>
        </li>
        <AdminOptions
          isAdmin={false}
          isSuperAdmin={false}
        />
        <li
          className="nav-item"
        >
          <Link
            className="nav-link"
            onClick={[Function]}
            replace={false}
            to="/"
          >
            Logout
          </Link>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;

exports[`<UserTopNav /> should render correctly for super admin user 1`] = `
<nav
  className="navbar navbar-expand-md navbar-dark fixed-top d-block d-lg-none"
>
  <div
    className="container"
  >
    <a
      className="navbar-brand"
      href=""
    >
      <strong
        className="text-capitalize"
      >
        test-name
      </strong>
      <br />
      <p
        className="h6"
      />
    </a>
    <button
      className="navbar-toggler"
      data-target="#navMenu"
      data-toggle="collapse"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navMenu"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <li
          className="nav-item"
        >
          <Link
            className="nav-link"
            replace={false}
            to="/events"
          >
            My events
          </Link>
        </li>
        <li
          className="nav-item"
        >
          <Link
            className="nav-link"
            replace={false}
            to="/addEvent"
          >
            Add Events
          </Link>
        </li>
        <li
          className="nav-item"
        >
          <Link
            className="nav-link"
            replace={false}
            to="/centers2"
          >
            Centers
          </Link>
        </li>
        <AdminOptions
          isAdmin={false}
          isSuperAdmin={true}
        />
        <li
          className="nav-item"
        >
          <Link
            className="nav-link"
            onClick={[Function]}
            replace={false}
            to="/"
          >
            Logout
          </Link>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
